streams-bootstrap-large-messages/src/main/java/com/bakdata/kafka/LargeMessageKafkaStreamsApplication.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 bakdata
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.bakdata.kafka;
+
+import java.util.function.Consumer;
+
+/**
+ * Kafka Streams Application that automatically removes files associated with {@code LargeMessageSerde}
+ */
+public abstract class LargeMessageKafkaStreamsApplication extends KafkaStreamsApplication {
+
+    @Override
+    protected void cleanUpRun(final CleanUpRunner cleanUpRunner) {
+        this.registerLargeMessageCleanUpHook(cleanUpRunner);
+        super.cleanUpRun(cleanUpRunner);
+    }
+
+    private void registerLargeMessageCleanUpHook(final CleanUpRunner cleanUpRunner) {
```

Comment:
Can we move this into the utility class? Then it wouldn't be necessary to use this subclass, which can be hard in some cases where you already have a custom super class

Reply:
Done. Do you think it makes sense to even have this super class?